pandas/core/series.py

```diff
@@ -2144,10 +2144,19 @@ def idxmin(self, axis=0, skipna=True, *args, **kwargs):
         >>> s.idxmin(skipna=False)
         nan
         """
+        delegate = self._values
         skipna = nv.validate_argmin_with_skipna(skipna, args, kwargs)
-        i = nanops.nanargmin(self._values, skipna=skipna)
-        if i == -1:
-            return np.nan
+
+        if isinstance(delegate, ExtensionArray):
+            if not skipna and isna(delegate).any():
+                return np.nan
+            else:
+                i = delegate.argmin()
+
+        else:
+            i = nanops.nanargmin(delegate, skipna=skipna)
```

Comment:
Maybe we could avoid repeating part of the code here by calling `self.argmin()` ?

Reply:
Thanks. Done.